pkg/virt-operator/resource/generate/components/daemonsets.go

```diff
@@ -22,12 +22,14 @@ const (
 	kubeletPodsPath = "/var/lib/kubelet/pods"
 )
 
-func NewHandlerDaemonSet(namespace string, repository string, imagePrefix string, version string, launcherVersion string, productName string, productVersion string, productComponent string, pullPolicy corev1.PullPolicy, migrationNetwork *string, verbosity string, extraEnv map[string]string) (*appsv1.DaemonSet, error) {
+func NewHandlerDaemonSet(namespace, repository, imagePrefix, version, launcherVersion, productName, productVersion, productComponent, image, launcherImage string, pullPolicy corev1.PullPolicy, migrationNetwork *string, verbosity string, extraEnv map[string]string) (*appsv1.DaemonSet, error) {
 
 	deploymentName := VirtHandlerName
-	imageName := fmt.Sprintf("%s%s", imagePrefix, deploymentName)
+	if image == "" {
+		image = fmt.Sprintf("%s%s", imagePrefix, deploymentName)
+	}
 	env := operatorutil.NewEnvVarMap(extraEnv)
-	podTemplateSpec, err := newPodTemplateSpec(deploymentName, imageName, repository, version, productName, productVersion, productComponent, pullPolicy, nil, env)
+	podTemplateSpec, err := newPodTemplateSpec(deploymentName, image, repository, version, productName, productVersion, productComponent, pullPolicy, nil, env)
```

Comment:
Probably you should also update `newPodTemplateSpec` function.
The container image will be defined as `<repository>/<image><version>`
https://github.com/kubevirt/kubevirt/blob/e24c029b9185341fffef24bd7894e1429788997e/pkg/virt-operator/resource/generate/components/deployments.go#L172
Since now image could be the entire string `some.registry.com:tag1`, the resulting image path is wrong.
Or am I missing something? Thanks

Reply:
Great catch!!! Thank you!

Reply:
I haven't seen the new changes yet, but what about adding a test that covers this case? what do you think?

Reply:
Sounds good, will do